stl/src/xlock.cpp

```diff
@@ -11,12 +11,12 @@
 
 _STD_BEGIN
 
-#define MAX_LOCK 8 // must be power of two
+constexpr int _Max_lock = 8; // must be power of two

```

Comment:
Pre-existing: Should we axe this comment and instead include `<bit>` and `static_assert(std::ispow2(_Max_lock));`?

Reply:
That might fail to compile; `<bit>` requires `/std:c++latest` mode, but our legacy build system compiles certain things as `/clr:pure /std:c++14` and `/clr /std:c++17` if I recall correctly. (This will be ported to the CMake build system in the future.) I recommend keeping the comment and making no additional changes here; there is no real safety to be gained by a `static_assert` as we aren’t changing this value and powers of two are easily recognizable (unlike, for example, an array being sorted).

Reply:
Thanks for the reminder.